tests/components/lovelace/test_dashboard.py

```diff
@@ -275,3 +293,154 @@
     assert response["result"] == {"hello": "yo2"}
 
     assert len(events) == 1
+
+
+async def test_storage_dashboards(hass, hass_ws_client, hass_storage):
+    """Test we load lovelace config from storage."""
+    assert await async_setup_component(hass, "lovelace", {})
+    assert hass.data[frontend.DATA_PANELS]["lovelace"].config == {"mode": "storage"}
+
+    client = await hass_ws_client(hass)
+
+    # Fetch data
+    await client.send_json({"id": 5, "type": "lovelace/dashboards/list"})
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"] == []
+
+    # Add a dashboard
+    await client.send_json(
+        {
+            "id": 6,
+            "type": "lovelace/dashboards/create",
+            "url_path": "created_url_path",
+            "require_admin": True,
+            "sidebar": {"title": "Updated Title", "icon": "mdi:map"},
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"]["require_admin"] is True
+    assert response["result"]["sidebar"] == {
+        "title": "Updated Title",
+        "icon": "mdi:map",
+    }
+
+    dashboard_id = response["result"]["id"]
+
+    assert "created_url_path" in hass.data[frontend.DATA_PANELS]
+
+    await client.send_json({"id": 7, "type": "lovelace/dashboards/list"})
+    response = await client.receive_json()
+    assert response["success"]
+    assert len(response["result"]) == 1
+    assert response["result"][0]["mode"] == "storage"
+    assert response["result"][0]["sidebar"] == {
+        "title": "Updated Title",
+        "icon": "mdi:map",
+    }
+    assert response["result"][0]["require_admin"] is True
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5093948)